File: velolabs_repos/skylock_ios/Skylock/Skylock/CrashHandler.swift
//
//  CrashHandler.swift
//  Ellipse
//
//  Created by Andre Green on 2/8/17.
//  Copyright © 2017 Andre Green. All rights reserved.
//

import Foundation

class CrashHandler: Accelerometer {
    override var maxPoints: Int {
        return 3
    }
    
    private let stdDevCutOff:Float = 323.0
    private let valueCuttOff:Float = 782.3
    
    override func shouldAlert() -> Bool {
        if points.count < maxPoints {
            return false
        }
        
        let stdDevs = allStdDevs()
        guard let xDev = stdDevs[.xValue], let yDev = stdDevs[.yValue], let zDev = stdDevs[.zValue] else {return false}
        
        let aves = average()
        guard let xAve = aves[.xValue], let yAve = aves[.yValue], let zAve = aves[.zValue] else {return false}
        
        print(xAve, yAve, zAve, xDev, yDev, zDev)
        if (check(value: xAve) || check(value: yAve) || check(value: zAve))
            && (checkStdDev(value: xDev) || checkStdDev(value: yDev) || checkStdDev(value: zDev))
        {
            print("Throwing crash alert with values: xAve: \(xAve) yAve: \(yAve) zAve: \(zAve) ")
            print("xDev: \(xDev), yDev: \(yDev), zDev: \(zDev)")
            points.removeAll()
            return true
        }
        
        return false
    }
    
    func checkStdDev(value: Float) -> Bool {
        return  value > stdDevCutOff
    }
    
    func check(value: Float) -> Bool {
        return value > valueCuttOff
    }
}

File: velolabs_repos/Skyfleet-Android/app/src/main/java/cc/skylock/skylock/bluetooth/CrashHandler.java
package cc.skylock.skylock.bluetooth;

import android.util.Log;

/**
 * Created by admin on 11/02/17.
 */

public class CrashHandler extends Accelerometer {
    String TAG = CrashHandler.class.getName();

    private float sensitivity;

    private double stdDevCutOff =323.0;
    private double valueCuttOff =782.3;

    CrashHandler(){
        super.maxPoints = 3;
    }

    @Override
    public boolean shouldAlert() {
        if(accelerometerValues.size() < maxPoints) {
            return false;
        }
        float[] stdDevs = allStdDevs();
        float xDev = stdDevs[0];
        float yDev = stdDevs[1];
        float zDev = stdDevs[2];

        float[] aves = average();
        float xAve = aves[0];
        float yAve = aves[1];
        float zAve = aves[2];

        if((check(xAve) || check(yAve) || check(zAve)) && (checkStdDev(xDev) || checkStdDev(yDev) || checkStdDev(zDev))){
            Log.i(TAG,"Throwing Crash alert with values Ave: x:"+xAve+",y:"+yAve+",z:"+zAve);
            Log.i(TAG,"Throwing Crash alert with values: Dev x:"+xDev+",y:"+yDev+",z:"+zDev);
            accelerometerValues.clear();
            return true;
        }
        return false;
    }

    public boolean checkStdDev(float value){
        return value  > stdDevCutOff;
    }

    public boolean check(float value){
        return value  > valueCuttOff;
    }
}
